src/eval.c

```diff
@@ -5394,6 +5465,779 @@ get_lit_string_tv(char_u **arg, typval_T *rettv, int evaluate)
+    {
+	length += STRLEN(embedded[i]->expr);
+    }
+
+    return length;
+}
+
+    static size_t
+sum_embedded_stringified_strlen(embedded_t** embedded, size_t embedded_size)
+{
+    size_t	length = 0;
+    size_t	i;
+
+    for (i = 0; i < embedded_size; ++i)
+    {
+	length += STRLEN(embedded[i]->stringified);
+    }
+
+    return length;
+}
+
+    static size_t
+sum_embedding_strlen(embedded_t** embedded, size_t embedded_size)
+{
+    const size_t	LITERAL_EMBEDDING_LENGTH = 3;  // a number of strlen("${}")
+    const size_t	VAR_EMBEDDING_LENGTH = 1;  // a number of strlen("$")
+    size_t		length = 0;
+    size_t		i;
+
+    for (i = 0; i < embedded_size; ++i)
+    {
+	length += embedded[i]->is_embedded_literal
+		? LITERAL_EMBEDDING_LENGTH
+		: VAR_EMBEDDING_LENGTH;
+    }
+
+    return length;
+}
+
+/*
+ * Count up independent "$"s on `template`.
+ * (Count up embedding.)
+ */
+     static size_t
+count_embedding(char_u *template)
+{
+    size_t	count = 0;
+    char_u*	p = template + 1;  // skip a head of '$'
+
+    // Allow to refer -1 at later
+    if (p[0] == '$' && p[1] != '$')
+    {
+	++count;
+    }
+    MB_PTR_ADV(p);
+
+    for (; *p != NUL; MB_PTR_ADV(p))
+    {
+	// Is `p` an independent '$'? e.g. "x$x". not "$$x" nor "x$$".
+	if (p[-1] != '$' && p[0] == '$' && p[1] != '$')
+	{
+	    ++count;
+	    ++p;      // a part of "$"
+	    continue; // a part of "{" (incresing)
+	}
+    }
+
+    return count;
+}
+
+/*
+ * Allocate `size + 1` memory,
+ * also set the last to NUL.
+ */
+    static char_u*
+alloc_terminated(size_t size)
+{
+    char_u*	p = alloc(size + 1);
+    p[size] = NUL;
+    return p;
+}
+
+/*
+ * Allocate a `embedded_t*`.
+ */
+    static embedded_t*
+new_embedded(int is_embedded_literal, char_u *expr, size_t expr_size)
+{
+    embedded_t *x = alloc(sizeof(embedded_t));
+    x->is_embedded_literal = is_embedded_literal;
+
+    x->expr = alloc_terminated(expr_size);
+    memcpy(x->expr, expr, expr_size);
+
+    x->stringified = NULL;
+    return x;
+}
+
+/*
+ * Free it and its fields.
+ */
+    static void
+delete_embedded(embedded_t** x)
+{
+    embedded_t *xx = *x;
+
+    free(xx->expr);
+    xx->expr = NULL;
+
+    if (xx->stringified != NULL)
+    {
+	free(xx->stringified);
+	xx->stringified = NULL;
+    }
+
+    free(*x);
+    *x = NULL;
+}
+
+/*
+ * Parse a `template` string into a (non template) string.
+ *
+ * The result MUST BE `free`d later! (if it is not NULL.)
+ */
+    static char_u*
+parse_template(
+	char_u* (*enclose)(char_u*),
+	char_u* (*disclose)(char_u*),
+	char_u* (*fix_quote)(char_u),
+	int (*is_closing_quote)(char_u*),
+	char_u *template,
+	int evaluate)
+{
+    const size_t	QUOTES_LENGTH = 3;  // a number of strlen("$''") (or strlen("$\"\""))
+    const size_t	OPNEING_QUOTE_LENGTH = 2;  // a number of strlen("$'")
+    size_t		i;
+    char_u*		result;
+    size_t		embedded_size;  // a size of embedded
+    embedded_t**	embedded = read_embedded(template, &embedded_size);
```

Comment:
Same for this `**`.

Reply:
I'll `%s/\v([a-zA-Z_]+)(\*+)([ 	])([\(a-zA-Z_])/\1\3\2\4/gc` :+1: